libportal/meson.build

```diff
@@ -222,16 +232,25 @@ if 'gtk4' in backends
       )
     endif
   endif
+  enabled_backends += ['gtk4']
 endif
 
 ########
 # Qt 5 #
 ########
 
-if 'qt5' in backends
+require_qt5 = 'qt5' in backends
+want_qt5 = 'auto' in backends or require_qt5
+have_cpp = false
+
+if want_qt5
+  have_cpp = add_languages('cpp', required: require_qt5)
+endif
+
+qt5_dep = dependency('qt5', modules: ['Core', 'Gui', 'X11Extras', 'Widgets'], required: require_qt5)
+if want_qt5 and have_cpp and qt5_dep.found()
   add_languages('cpp', required : true)
```

Comment:
You now have add_languages being done twice, but it's redundant here.

Reply:
fixed, thanks!